### ISSUE TITLE
Orchestrator might stop accepting work when it is waiting for a ticket redemption to confirm

### ISSUE THREAD

**Issue by yondonfu** [livepeer/go-livepeer#1590]
**Describe the bug**

Note: There are actually a few more requirements for this scenario to occur that I'll add in follow up comments later. 

Currently, an orchestrator (O) stops accepting work in the following scenario:

Let a broadcaster's (B) reserve allocation for the orchestrator be X.

1. O sets face value of tickets to X to match the B's reserve allocation
2. B's max float is currently X
3. O receives a winning ticket and redeems it
4. B's max float is now 0 because there X amount of value considered as pending until the redemption tx confirms on-chain
5. O calculate the new ticket params to send back to B
6. O sees that B's max float is 0 so it returns an `insufficient sender reserve` error to B
7. Step 6 repeats until O sees that the redemption tx confirms on-chain

When the above scenario occurs, the orchestrator logs will look like this:

```
I0729 18:44:46.518036   93856 backend.go:96]
******************************Eth Transaction******************************

Invoking transaction: "redeemWinningTicket". Inputs: "_ticket: { Recipient: 0x6788F9AAfEACD5e31559EAaD31fd785DE034260e  Sender: 0x8A8d7Bee38783dfe5568a64Dd50dbceAb58F1dBe  FaceValue: 500000000000000000  WinProb: 23158417847463239084714197001737581570653996933128112807891600000000000000000  SenderNonce: 2  RecipientRandHash: 0x5483875c46c23a6f4be472727d2c737cd91076d922ee2171d22e6edc554e263e  AuxData: 0x000000000000000000000000000000000000000000000000000000000000004783d8f1c512151560453e2f6f3e990d8c2725db83b2888ca5321d569770f8cc8d }  _sig: 0xa76731de83e5999f4ad123b9d22c0e7891156b0dec1d0c89351b6d7d61d983790bd0ca9b265c355f6d0228a130fffae18b062c765e8e375ab1fb7193477148091b  _recipientRand: 2655046112346169819941714599222515298333337682031180840906545582045193452504"  Hash: "0xdd096a25664113c75084935056c656c74bc27371e8a4191cb392127159d4be39".

***************************************************************************
I0729 18:44:47.340410   93856 gaspricemonitor.go:134] Cached gas price: 20000000000
I0729 18:44:47.818960   93856 orchestrator.go:532] Transcoding of segment manifestID=4772b69c seqNo=12 took=2.782608018s
I0729 18:44:47.818994   93856 orchestrator.go:547] Transcoded segment manifestID=4772b69c seqNo=12 profile=P240p30fps16x9 len=201724
I0729 18:44:47.820103   93856 orchestrator.go:547] Transcoded segment manifestID=4772b69c seqNo=12 profile=P360p30fps16x9 len=293844
I0729 18:44:47.822847   93856 player.go:105] LPMS got HTTP request @ /stream/4772b69c/P240p30fps16x9/12.ts
I0729 18:44:47.822862   93856 player.go:105] LPMS got HTTP request @ /stream/4772b69c/P360p30fps16x9/12.ts
I0729 18:44:48.340974   93856 gaspricemonitor.go:134] Cached gas price: 20000000000
E0729 18:44:48.374584   93856 segment_rpc.go:88] Error updating orchestrator info - err=insufficient sender reserve
```

Note the last log statement which indicates that the broadcaster's reserve is insufficient because the broadcaster's max float is 0.

The broadcaster logs will look like this:

```
Trying to transcode segment nonce=3822204207721735489 seqNo=12
I0729 18:44:45.024911   93846 segment_rpc.go:682] Created new payment - manifestID=4772b69c recipient=0x6788F9AAfEACD5e31559EAaD31fd785DE034260e faceValue=0.5 ETH winProb=0.2000000000 price=5050000000.000 wei/pixel numTickets=2
I0729 18:44:45.025031   93846 segment_rpc.go:382] Submitting segment nonce=3822204207721735489 manifestID=4772b69c seqNo=12 bytes=528844 orch=https://127.0.0.1:8936
I0729 18:44:45.035622   93846 segment_rpc.go:416] Uploaded segment nonce=3822204207721735489 manifestID=4772b69c seqNo=12 orch=https://127.0.0.1:8936 dur=10.555521ms
I0729 18:44:46.750934   93846 broadcast.go:288] Processing segment nonce=3822204207721735489 manifestID=4772b69c seqNo=13 dur=1.98
I0729 18:44:46.750971   93846 broadcast.go:360] No sessions available for segment nonce=3822204207721735489 manifestID=4772b69c seqNo=13
I0729 18:44:46.750999   93846 broadcast.go:130] Starting session refresh manifestID=4772b69c
I0729 18:44:46.751325   93846 rpc.go:223] Connecting RPC to https://127.0.0.1:8936
E0729 18:44:46.756965   93846 rpc.go:215] Could not get orchestrator orch=https://127.0.0.1:8936 err=rpc error: code = Unknown desc = insufficient sender reserve
```

Note the last log statement which indicates that the orchestrator is refusing to connect with the broadcaster because the broadcaster's max float is 0.

I think the original rationale for the current behavior was that it was the most conservative/defensive behavior for O - the current strategy ensures that O will never receive a winning ticket with a face value that is not fully covered by the B's reserve allocation. However, the problem with the above is that O stops accepting work (meaning that the B could never come back to the O!) even if B has a large deposit. So, this doesn't seem like the right tradeoff to make. There is probably an alternative design where the O takes on slightly more risk that the ticket face value is not fully covered by the B's reserve allocation, but will continue to accept work in this scenario even if it is waiting for a ticket redemption to confirm on-chain. One possibility is for O to also look at B's deposit and if it is sufficient (maybe some multiple of the reserve allocation or the tx cost of redemption) then O will continue accepting work.

Rough sketch of an idea around a possible approach to get thinking started:

If B's deposit > B's reserve allocation * a constant factor, then use B's reserve allocation as the max float even if there is pending value (i.e. a pending winning ticket redemption tx).

Note that the above is even more of a problem when B is working with multiple Os associated with the same on-chain address because when a winning ticket is sent *all* Os will stop accepting work until that redemption tx confirms on-chain.

**Comment by yondonfu** [comment]
Below is a spec for a proposed solution:

# Minimum Deposit Float Ratio

## Abstract


An orchestrator can define a **minimum deposit float ratio** which is used to determine whether a broadcaster's deposit is sufficiently high for the orchestrator to take on the risk of ignoring the broadcaster's current float when calculating the broadcaster's max float.

## Motivation


For background:

- A broadcaster's reserveAlloc is the value from the broadcaster's reserve committed to an orchestrator. This can be calculated at the beginning of a round as reserve / # of active orchestrators and is fixed for the duration of a round
- A broadcaster's float is the value of winning tickets sent by the broadcaster and received by an orchestrator that have not yet been successfully redeemed on-chain
- A broadcaster's max float is the maximum value that an orchestrator is willing to float the broadcaster at any point in time. When the broadcaster's float increases, the broadcaster's max float decreases. When the broadcaster's float decreases, the broadcaster's max float increases
- A broadcaster's max float is currently calculated as `reserveAlloc - float` [1]

[1] Note that float is currently referred to as `pendingAmount` in the code.

See the OP in this thread for details on the specific problem that we're trying to solve for. 

TLDR: The current calculation approach for a broadcaster's max float can cause an orchestrator to stop accepting work when it receives a winning ticket which represents value lost for the orchestrator.

## Proposed Solution


Define a constant `minDepositFloatRatio`. We may want to allow this value to be configurable in the future, but given that it is difficult to reason about what a reasonable value would be I'm leaning toward just defining this as a constant for now.

Change the calculation approach for a broadcaster's max float to:

- `depositFloatRatio = deposit / float`
- If `depositFloatRatio >= minDepositFloatRatio`:
    - Return `reserveAlloc`
- Else:
    - Return `reserveAlloc - float`

`depositFloatRatio` serves as a measure of how much greater a broadcaster's deposit is than the broadcaster's current float. The idea is to have the orchestrator take on the risk of ignoring the broadcaster's current float when calculating the broadcaster's max float if the ratio is sufficiently high. An initial value for `minDepositFloatRatio` might be 10.0 meaning that the broadcaster's deposit needs to be 10x the broadcaster's float in order for the orchestrator to ignore the broadcaster's float during the max float calculation.

## Implementation Tasks and Considerations


- Update [maxFloat()](https://github.com/livepeer/go-livepeer/blob/82509ae41b822a3ab54e266315eb6a9dd0a692ff/pm/sendermonitor.go#L175) method defined in `pm.SenderMonitor` to use the calculation approach described above

## Testing Tasks and Considerations


- Update unit tests
- Make sure that the scenario described in the OP is no longer an issue
